src/main/java/seedu/address/model/task/TaskList.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.task.exceptions.TaskNotFoundException;
+
+/**
+ * A list of tasks that does not allow nulls.
+ *
+ * Supports a minimal set of list operations.
+ */
+public class TaskList implements Iterable<Task> {
+
+    private final ObservableList<Task> internalList = FXCollections.observableArrayList();
```

Comment:
Very good use of ObservableList, this will help immensely in the integration into the UI!